lumicks/pylake/tests/test_imaging_confocal/test_kymo.py

```diff
@@ -372,18 +372,58 @@ def get_rectangle_data():
         kymo.plot_with_time_histogram(color_channel="red", pixels_per_bin=6)
 
 
-def test_save_tiff(tmpdir_factory, test_kymos):
+def test_export_tiff(tmp_path, test_kymos):
     from os import stat
 
-    tmpdir = tmpdir_factory.mktemp("pylake")
+    def grab_tags(file):
+        import tifffile
+        from ast import literal_eval
+
+        tiff_tags = []
+        with tifffile.TiffFile(file) as tif:
+            for page in tif.pages:
+                page_tags = {}
+                for tag in page.tags.values():
+                    name, value = tag.name, tag.value
+                    try:
+                        page_tags[name] = literal_eval(value)
+                    except (ValueError, SyntaxError):
+                        page_tags[name] = value
+                tiff_tags.append(page_tags)
+        return tiff_tags
 
     kymo = test_kymos["Kymo1"]
-    kymo.export_tiff(f"{tmpdir}/kymo1.tiff")
-    assert stat(f"{tmpdir}/kymo1.tiff").st_size > 0
+    kymo.export_tiff(tmp_path / "kymo1.tiff")
+    assert stat(tmp_path / "kymo1.tiff").st_size > 0
+
+    # Check if tags were properly stored, i.e. test functionality of `_tiff_image_metadata()`,
+    # `_tiff_timestamp_ranges()` and `_tiff_writer_kwargs()`
+    tiff_tags = grab_tags(tmp_path / "kymo1.tiff")
+    assert len(tiff_tags) == 1
+    for tags, timestamp_range in zip(tiff_tags, kymo._tiff_timestamp_ranges()):
+        assert tags["ImageDescription"] == kymo._tiff_image_metadata()
+        assert tags["DateTime"] == f"{timestamp_range[0]}:{timestamp_range[1]}"
+        assert tags["Software"] == kymo._tiff_writer_kwargs()["software"]
+        np.testing.assert_allclose(
+            tags["XResolution"][0] / tags["XResolution"][1],
+            kymo._tiff_writer_kwargs()["resolution"][0],
+            rtol=1e-1,
+        )
+        np.testing.assert_allclose(
+            tags["YResolution"][0] / tags["YResolution"][1],
+            kymo._tiff_writer_kwargs()["resolution"][1],
+            rtol=1e-1
+        )
```

Comment:
why does the acceptance tolerance have to be so high here? that's a bit concerning

Reply:
The "XResolution" and "YResolution" are stored as a rational datatype with two 32 bit integers. This means these tags only give approximate resolution in many cases and the sole purpose (I can see) of these is solely for qualitative scaling and displaying purposes. Therefore, I also included the `pixel_size_um` in the ImageDescription tag (i.e. the `_tiff_image_metadata()` method). Still, I also included the resolution tags, as these were also set by the old `save_tiff()` function.
Note: The old tests did not take into account that the resolution tags were stored as rational, and only tested for the numerator.

Reply:
yeah, I kinda suspected it had something to do with the stored datatype. that's fine then, we don't really rely on this tag anyway and with the `pixel_size_um` there that's good